src/main/java/org/sagebionetworks/bridge/spring/controllers/CRCController.java

```diff
@@ -0,0 +1,381 @@
+                .build();
+        
+        sessionUpdateService.updateParticipant(session, context, updatedParticipant);
+        
+        return UserSessionInfo.toJSON(session);
+    } 
+    
+    @PutMapping("/v1/cuimc/appointments")
+    public ResponseEntity<StatusMessage> postAppointment() {
+        httpBasicAuthentication();
+        
+        IParser parser = FHIR_CONTEXT.newJsonParser();
+        JsonNode data = parseJson(JsonNode.class);
+        Appointment appointment = parser.parseResource(Appointment.class, data.toString());
+        
+        String userId = findUserId(appointment.getIdentifier());
+        
+        int status = writeReportAndUpdateState(userId, data, APPOINTMENT_REPORT, TESTS_SCHEDULED);
+        if (status == 200) {
+            return ResponseEntity.ok(new StatusMessage("Appointment updated."));
+        }
+        return ResponseEntity.created(URI.create("/v1/cuimc/appointments/" + userId))
+                .body(new StatusMessage("Appointment created."));
+    }
+
+    @PutMapping("/v1/cuimc/procedurerequests")
+    public ResponseEntity<StatusMessage> postProcedureRequest() {
+        httpBasicAuthentication();
+        
+        IParser parser = FHIR_CONTEXT.newJsonParser();
+        JsonNode data = parseJson(JsonNode.class);
+        ProcedureRequest procedure = parser.parseResource(ProcedureRequest.class, data.toString());
+        
+        String userId = findUserId(procedure.getIdentifier());
+        
+        int status = writeReportAndUpdateState(userId, data, PROCEDURE_REPORT, AccountStates.TESTS_COLLECTED);
+        if (status == 200) {
+            return ResponseEntity.ok(new StatusMessage("ProcedureRequest updated."));
+        }
+        return ResponseEntity.created(URI.create("/v1/cuimc/procedurerequests/" + userId))
+                .body(new StatusMessage("ProcedureRequest created."));
+    }
+
+    @PutMapping("/v1/cuimc/observations")
+    public ResponseEntity<StatusMessage> postObservation() {
+        httpBasicAuthentication();
+        
+        IParser parser = FHIR_CONTEXT.newJsonParser();
+        JsonNode data = parseJson(JsonNode.class);
+        Observation observation = parser.parseResource(Observation.class, data.toString());
+        
+        String userId = findUserId(observation.getIdentifier());
+        
+        int status = writeReportAndUpdateState(userId, data, OBSERVATION_REPORT, AccountStates.TESTS_AVAILABLE);
+        if (status == 200) {
+            return ResponseEntity.ok(new StatusMessage("Observation updated."));
+        }
+        return ResponseEntity.created(URI.create("/v1/cuimc/observations/" + userId))
+                .body(new StatusMessage("Observation created."));        
+    }
+    
+    void createLabOrder(StudyParticipant participant) {
+        Patient patient = createPatient(participant);
+        
+        IParser parser = FHIR_CONTEXT.newJsonParser();
+        
+        System.out.println(parser.encodeResourceToString(patient));
```

Comment:
Does this print statement leak PHI into the logs?

Reply:
Definitely, this has to be replaced with a call to the external service... I will remove it, and this controller is not complete to deploy until we get information from them on how to post this patient record.